.github/workflows/go.yaml

```diff
@@ -0,0 +1,49 @@
+name: Go
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  basic-checks:
+    name: Run basic-checks
+    runs-on: ubuntu-latest
+    steps:
+    - name: Set up Go 1.x
+      uses: actions/setup-go@v2
+      with:
+        go-version: ^1.14
+      id: go
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: Get dependencies
+      run: |
+        go get -v -t -d ./...
+        if [ -f Gopkg.toml ]; then
+            curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+            dep ensure
+        fi
+
+    - name: go-fmt
+      run: go fmt ./...
+    
+    - name: go-vet
+      run: go vet ./...
+
+  integration-tests:
+    name: Run integration-tests
+    needs: basic-checks
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+       
+    - name: Build Test Image
+      run: docker build -t secret-agent-tester -f Dockerfile.testing .
+
+    - name: Test
+      run: docker run -i --rm -v ${GITHUB_WORKSPACE}:/root/go/src/github.com/ForgeRock/secret-agent secret-agent-tester make test
```

Comment:
If we're going to automate make here, it means we can't read this file to fully know what this action does, we have to go look at the Makefile to find out what this action really does.

More importantly, since `make test` runs generate and manifests, if running those fix any errors, it will hide the fact that the developer forgot to run them, making the tests pass when they shouldn't.

Personally I think the action should separately run `go vet ./...`, then `go test ./... -tags=integration -cover`, and that's it. `go fmt` will only modify files whose modifications will then go unnoticed, and same with `controller-gen`.

Reply:
Slight modification, `go test` automatically runs `go vet`, so all we need is the one action of running `go test`.

Reply:
that's a good point. However, I still there's a benefit of running the generate stuff. That way we can catch problems with `//+kubebuilder` tags

However, I do agree with go fmt. We can kill that.

Reply:
> More importantly, since make test runs generate and manifests, if running those fix any errors, it will hide the fact that the developer forgot to run them, making the tests pass when they shouldn't.

oof good point

Reply:
That's also a good point. What if we either A) run the controller-gen commands after go test, or B) use `git status` or similar to ensure nothing has changed after running the generation commands.

Or maybe both A and B.

Reply:
Actually, of the two options, I think B is the only proper solution, A is not good enough. Maybe someone else has another solution.

Reply:
I will experiment with both A and B. 
I think we'll benefit from having a "quick" `go test` before building Dockerfile.testing + integration tests. Yes, technically tests run twice as of now, but will fail quickly if there's an import problem, etc.

Reply:
👍 
Now it checks for any output from `git status` after `go fmt` and kubebuilder related generators.